### PR TITLE
[ENH] add tests and proper documentation for refactored activation inputs in TSC and TSR

### LINKED ISSUE #10354: [ENH] add tests and proper documentation for refactored `activation` inputs in TSC and TSR
PR https://github.com/sktime/sktime/pull/9622 refactors the `activation` and similar arguments in TSC and TSR to accept arbitrary class strings from `torch`, or classes.

However:
* this is not properly documented. The user has no idea what are valid inputs to the `activation` parameters.
* this is not properly tested. There are no tests checking the various options by which activations can now be passed.

I have separated this from the PR #9622 to avoid regression on the feature itself, and avoid having a PR that is open for too long.

I think @Faakhir30 should take this on, given that he worked on #9622. Please comment so I can assign.

### PR DESCRIPTION
**Pull request by Faakhir30**
#### Reference Issues/PRs

Fixes #10354 

#### What does this implement/fix? Explain your changes.

- tests supported activation types using get_test_params in `InceptionTimeClassifier`  
- improve docstring for `_instantiate_activation`
- fix activation calling according to #9622 in `contran` classifier and regressor

#### Did you add any tests for the change?

No

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://github.com/all-contributors/all-contributors/blob/master/docs/emoji-key.md)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

**Comment by Faakhir30** [comment]
qouted text is from issue description of #10354 

> this is not properly documented. The user has no idea what are valid inputs to the activation parameters.

we do have supported types mentioned in the docstring of end estimators, docstring also mentions recomended activations, so, I'm unsure how to improve this further for end-user, please leave a hint

> this is not properly tested. There are no tests checking the various options by which activations can now be passed.

for testing, is triggering all kinds of supported types of activation via `get_test_params` enough? like we did for metrics
Or we need a seperate tests? I think that `get_test_params` approach does provide coverage for all cases (except error ones).

**Comment by fkiraly** [comment]
> we do have supported types mentioned in the docstring of end estimators, docstring also mentions recomended activations, so, I'm unsure how to improve this further for end-user, please leave a hint

I mean, in the docstrings of the classes.

A full description of a parameter always needs to include: name, type, default; if not all values of the type are permitted, an exact description of which values are permitted. Then what different values encode, or how they exactly change behaviour.

> Or we need a seperate tests? I think that get_test_params approach does provide coverage for all cases (except error ones).

I would add a test that tests directly the coercion method, or selecting one class and going in isolation through the different possible input types.

**Comment by Faakhir30** [comment]
Apologies for asking again. Since this change would need to be propagated across 15+ files, could you please confirm whether the following pattern is what you're looking for in the parameter documentation:

```python
"""
activation : str, Callable, or None, default=None
    Activation applied to the output layer.

    Permitted values:
    - ``None``: no activation is applied. Required when using ``CrossEntropyLoss``.
    - ``str``: name of a class in ``torch.nn``.  Case-sensitive names are recommended
      and must match PyTorch (e.g., ``"ReLU"``, ``"LeakyReLU"``).
      Lowercase aliases for common activations are also accepted
      (e.g., ``"relu"`` is resolved to ``"ReLU"``).
      The class is instantiated with default constructor arguments.
      Must be a valid ``torch.nn`` activation; see
      https://pytorch.org/docs/stable/nn.html#non-linear-activations-weighted-sum-nonlinearity
    - ``torch.nn.Module``: a pre-instantiated module (e.g., ``torch.nn.ReLU()``).
      Arbitrary callables are not supported.

   Recommended: 'ReLU', 'Tanh', 'Sigmoid', 'LeakyReLU', 'ELU', 'SELU', 'GELU'
    
"""

```

> or selecting one class and going in isolation through the different possible input types.

yes, this is done using `InceptionTimeClassifier` in this PR.

**Comment by fkiraly** [comment]
> could you please confirm whether the following pattern is what you're looking for in the parameter documentation:

Yes, exactly that!

I only have minor suggestions for improvement:
* in bullet point lists, make sure they are surrounded by empty lines, otherwise they will not display properly in the docs (see preview)
* `None` should be explained better, because what does it mean "required for cross entropy loss, and there is no activation" - without activation function, there is no sensible network? There is always an activation or there is no output.
* ``torch.nn.Module`` -> "an instance of a ``torch.nn.Module`` subclass, for example ..." - this is clearer

**Review by fkiraly — APPROVED** [review]
Great! Docstring looks good - I checked a few and assume it is simply the same copy-pasted across all estimators. Test calses also make sense.
